src/os/filestore/FileJournal.cc

```diff
@@ -134,6 +138,7 @@ int FileJournal::_open(bool forwrite, bool create)
 
  out_fd:
   VOID_TEMP_FAILURE_RETRY(::close(fd));
+  fd = -1;
```

Comment:
@tchaikov This one line is really the fix. but the other changes improve the error output.